gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixJobScheduler.java

```diff
@@ -174,6 +182,15 @@ public NonScheduledJobRunner(Properties jobConfig, JobListener jobListener) {
     public void run() {
       try {
         GobblinHelixJobScheduler.this.runJob(this.jobConfig, this.jobListener);
+
+        // remove non-scheduled job catalog once done so it won't be re-executed
+        if (GobblinHelixJobScheduler.this.jobCatalog != null) {
```

Comment:
Yes, the job will be re-executed once the master comes up and will be removed once completed.